File: seminar_2/task_4/Ishmuratov.Rail/Task_4.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n = 8, x = 1, y = 1;
    int l = 0, r = (n-1) * max(x, y), mid = (l + r)/2;
    while (l + 1 < r) {
        mid = (l + r)/2;
        if(mid/x + mid/y < n-1) {
            l = mid;
        }
        else{
            r = mid;
        }
    }
    cout<<r + min(x, y)<<"\n";
}

File: seminar_2/task_4/gukasian.vladimir/task4.go
package main 

import "fmt"

func copyTime(n , x, y int) int {
    mid := 0
    left   := 0
    right  := (n-1) * max(x, y)
    
    for ; left + 1 < right ; {
        mid = (left + right) / 2

        if (mid / x + mid / y) < n - 1 {
            left = mid
        } else {
            right = mid
        }
    }
    
    return right + min(x, y)
}

func main() {
    fmt.Println(copyTime(5, 1, 2))
}
